ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/HardFork/Combinator/Protocol.hs

```diff
@@ -112,6 +115,13 @@ instance CanHardFork xs => ConsensusProtocol (HardForkProtocol xs) where
   -- Security parameter must be equal across /all/ eras
   protocolSecurityParam = hardForkConsensusConfigK
 
+  protocolSecurityParamConsistencyCheck HardForkConsensusConfig {..} = do
+    let allSecurityParams = hardForkConsensusConfigK :|
```

Comment:
Minor: `NonEmpty` is somewhat overkill; `1 < Data.Set.size (Data.Set.fromList (...))` would suffice, I think. (and avoid `nub`)

Reply:
Err, just `or (hardFOrkConsensusConfigK /=) $ perEraConsensusConfigSecurityParams hardForkConsensusConfigPerEra` actually.

Reply:
collecting the security params allows us (or node, whoever) to spit out the actual values of k in error messages, which is nice (and in theory allows HFC-style multi-type blocks with more than one level of nesting, which I don't think is actually useful)